test/functional/rpc_users.py

```diff
@@ -84,6 +94,39 @@ def test_auth(self, node, user, password):
         self.log.info('Wrong...')
         assert_equal(401, call_with_auth(node, user + 'wrong', password + 'wrong').status)
 
+    def test_rpccookieperms(self):
+
+        if os.name == 'nt':
+            raise SkipTest(f"Skip cookie file permissions checks as OS detected as: {os.name=:}")
+        self.log.info('Check cookie file permissions can be set using -rpccookieperms')
+
+        cookie_file_path = self.nodes[1].chain_path / '.cookie'
+
+        def test_perm(perm: Optional[str]):
+            if not perm:
+                perm = f"{(DEFAULT_COOKIE_PERMISSION & PERM_BITS_UMASK):03o}"
+                self.restart_node(1)
+            else:
+                self.restart_node(1, extra_args=[f"-rpccookieperms={perm}"])
+            file_stat = os.stat(cookie_file_path)
+            actual_perms = f"{(file_stat.st_mode & PERM_BITS_UMASK):03o}"
+            assert_equal(perm, actual_perms)
+
+       # Remove any leftover rpc{user|password} config options from previous tests
+        conf = self.nodes[1].bitcoinconf
+        with conf.open('r') as file:
+            lines = file.readlines()
+        filtered_lines = [line for line in lines if not line.startswith('rpcuser') and not line.startswith('rpcpassword')]
+        with conf.open('w') as file:
+            file.writelines(filtered_lines)
+
+        self.log.info('Check default cookie permission')
+        test_perm(None)
+
+        self.log.info('Check custom cookie permissions')
+        test_perms = ["440", "640", "444"]
+        [test_perm(perm) for perm in test_perms]
```

Comment:
Feels like a hack

```suggestion
        for perm in "440", "640", "444":
            test_perm(perm)
```

Reply:
IMO list comprehensions are fine, but seeing as I voided half the point of using one (creating a one-liner), so took the change to a regular for loop in 7516e9120e202395dabebd5b7212d9c2d3787c43